Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -501,6 +501,25 @@ final class PurchasesOrchestrator {
         return (transaction, customerInfo, userCancelled)
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    private func purchase(
+        _ product: SK2Product,
+        _ options: Set<Product.PurchaseOption>
+    ) async throws -> Product.PurchaseResult {
+        // Note: this can be simplified as `#if swift(>=5.9) && os(xrOS)`
+        // once we drop support for Xcode 13.x
```

Comment:
Short-circuiting doesn't work in Xcode 13: https://twitter.com/nachosoto/status/1671666952773971968?s=61&t=eV60wxp04gdupmhWzuBI1g